Amazon.IonObjectMapper.Test/TestObjects.cs

```diff
@@ -1,829 +1,841 @@
-            this.PrivateProperty = privateValue;
-            this.PrivateProtectedProperty = privateProtectedValue;
-        }
-
-        public string PublicProperty { get; set; }
-        protected string ProtectedProperty { get; set; }
-        protected internal string ProtectedInternalProperty { get; set; }
-        internal string InternalProperty { get; set; }
-        private string PrivateProperty { get; set; }
-        private protected string PrivateProtectedProperty { get; set; }
-
-        public override string ToString()
-        {
+﻿/*
+ * Copyright (c) Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace Amazon.IonObjectMapper.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+
+    [IonAnnotateType]
+    public class Vehicle
+    {
+        public override string ToString()
+        {
+            return "<Vehicle>";
+        }
+    }
+
+    [IonAnnotateType(Prefix = "my.universal.namespace", Name="BussyMcBusface")]
+    public class Bus : Vehicle
+    {
+        public override string ToString()
+        {
+            return "<Bus>";
+        }
+    }
+
+    public class Truck : Vehicle
+    {
+        public override string ToString()
+        {
+            return "<Truck>";
+        }
+    }
+
+    public class Plane : Vehicle
+    {
+        public int MaxCapacity { get; init; }
+
+        public override string ToString()
+        {
+            return "<Plane>";
+        }
+    }
+
+    [IonAnnotateType]
+    public class Boat : Vehicle
+    {
+        public override string ToString()
+        {
+            return "<Boat>";
+        }
+    }
+
+    [IonDoNotAnnotateType(ExcludeDescendants = true)]
+    public class Motorcycle : Vehicle
+    {
+        public string Brand { get; init; }
+
+        [IonField]
+        public string color;
+
+        [IonField]
+        public bool canOffroad;
+
+        public override string ToString()
+        {
+            return "<Motorcycle>{ Brand: " + Brand + ", color: " + color + ", canOffroad: " + canOffroad + " }";
+        }
+    }
+
+    [IonDoNotAnnotateType(ExcludeDescendants = true)]
+    public class Yacht : Boat
+    {
+        public override string ToString()
+        {
+            return "<Yacht>";
+        }
+    }
+
+    [IonDoNotAnnotateType(ExcludeDescendants = true)]
+    public class Ship : Boat
+    {
+        public string Name { get; init; }

```

Comment:
The whole idea of this `Ship` class was to test case insensitive comparisons so the weird casing here was entirely deliberate. Is there a way to change it back without StyleCop complaining about it?

Reply:
Nevermind I misunderstood the diff (I now see `ShipWithVariedCasing` below)